jsonschema-core.xml

```diff
@@ -1830,6 +1820,74 @@
                 </section>
             </section>
 
+            <section title="Compound Documents">
+                <t>
+                    A Compound Schema Document is defined as JSON document (sometimes called a "bundled" schema)
+                    which has multiple embedded JSON Schema Resources bundled into the same document to
+                    ease transportation.
+                </t>
+                <t>
+                    Each embedded Schema Resource MUST be treated as an individual Schema Resource, following standard
+                    schema loading and processing requirements, including determining voacbulary support.
+                </t>
+                <section title="Bundling">
+                    <t>
+                        A Compound Schema Document is created by taking references (such as "$ref")
+                        to an external Schema Resource and embedding the referenced Schema Resources
+                        within the referreing document.
+                    </t>
+                    <t>
+                        Each embedded JSON Schema Resource (and the document's root) MUST identify itself
+                        with an absolute URI using the "$id" keyword, and SHOULD make use of the "$schema"
+                        keyword to identify the dialect it is using, in the root of the schema resource.
+                    </t>
+                    <t>
+                        When a by-reference applicator is dereferenced as part of the bundling process, the Schema Resource
+                        MUST NOT be embedded by replacing the schema object from which it was referenced, or by wrapping
+                        in other applicator keywords. Instead, a bundled Schema Resource MUST be located as a value of
+                        a "$defs" object at the containing schemas root. The key of the "$defs" for the now embedded
+                        Schema Resource MAY be the "$id" of the bundled schema or some other form of application defined
+                        unique identifer (such as a UUID). This key is not intended to be referenced in JSON Schema, but may
+                        be used by an application to aid the bundling process.
+                    </t>
+                    <t>
+                        References in the containing schema document to the previously external Schema Resources
+                        MUST NOT be changed, and now resolve to a schema using the "$id" of an embedded Schema Resource.
```

Comment:
MUST NOT because you want to preserve the original reference to the Schema Resource (for the potential resulting output), as opposed to using a new one making use of `$defs` in the documents root.

Reply:
In that case I think the conditional "In order to produce identical output..." or something like that needs to be attached to the MUST NOT.  Or be made very clear in the framing of this section (which maybe it is and I've just lost track of that?)

Reply:
Additionally, there was some previous talk (I forget who from specifically) about doing this in a way where the process would change all the references, done in such a way which could potentially cause failures in dereferencing later.
So it's not JUST for preserving output.

Reply:
If you use the "output" wording, please make it clear that this includes not just evaluation success/failure, but also all the uris and paths used in the resulting annotations or errors.